src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/ErrorHandlingTransmissionPolicy.cs

```diff
@@ -63,47 +75,141 @@ private void HandleTransmissionSentEvent(object sender, TransmissionProcessedEve
-                            this.pauseTimer.Start(() =>
-                            {
-                                this.MaxBufferCapacity = null;
-                                this.MaxSenderCapacity = null;
-                                this.LogCapacityChanged();
-                                this.Apply();
-
-                                return TaskEx.FromResult<object>(null);
-                            });
-
-                            this.Transmitter.Enqueue(e.Transmission);
-                            break;
-                    }
+                    this.HandleStatusCode((int)httpWebResponse.StatusCode, e.Transmission);
                 }
-                else 
+                else
                 {
                     TelemetryChannelEventSource.Log.TransmissionSendingFailedWebExceptionWarning(e.Transmission.Id, webException.Message, (int)HttpStatusCode.InternalServerError);
                 }
             }
             else
             {
-                var theException = e.Exception as Exception;
-                if (theException != null)
+                if (!string.IsNullOrEmpty(e.ResponseContent))
+                {
+                    // Partial success case (206)
+                    var newTransmissions = this.ParsePartialSuccessResponse(e.Transmission, e.ResponseContent);
+
+                    if (newTransmissions != null)
+                    {
+                        foreach (int statusCode in newTransmissions.Keys)
+                        {
+                            byte[] data = JsonSerializer.ConvertToByteArray(newTransmissions[statusCode]);
+                            Transmission newTransmission = new Transmission(
+                                e.Transmission.EndpointAddress,
+                                data,
+                                e.Transmission.ContentType,
+                                e.Transmission.ContentEncoding,
+                                e.Transmission.Timeout);
+
+                            this.HandleStatusCode(statusCode, newTransmission);
+                        }
+                    }
+                }
+                else
                 {
-                    TelemetryChannelEventSource.Log.TransmissionSendingFailedWarning(e.Transmission.Id, theException.Message);
+                    if (e.Exception != null)
+                    {
+                        TelemetryChannelEventSource.Log.TransmissionSendingFailedWarning(e.Transmission.Id, e.Exception.Message);
+                    }
+
+                    this.ConsecutiveErrors = 0;
                 }
+            }
+        }
 
+        private IDictionary<int, string> ParsePartialSuccessResponse(Transmission initialTransmission, string response)
+        {
+            BreezeResponse breezeResponse;
+            try
+            {
+                breezeResponse = new JavaScriptSerializer().Deserialize<BreezeResponse>(response);
```

Comment:
store a static instance of JavaScriptSerializer as variable to reuse every time instead of create a new instance for every call?